### PR TITLE
Update box create step to wait for a successful create operation before allowing the HCP Vagrant post-processor to continue

### LINKED ISSUE #128: Vagrant-registry post-processor failure on box creation
####  Debug output

Screenshot from 2024-08-16 12-40-55

Screenshot from 2024-08-16 12-56-33

### Expected behavior

Package.box file should have been uploaded to the Hashicorp Vagrant Registry using the Packer post-processor "vagrant-registry".

### Actual behavior

The box was "created" but the package.box file never got uploaded to the regsitry.

#### Vagrant version

Vagrant 2.4.1

#### Packer version

Packer v1.11.2

#### Host operating system

Ubuntu 22.04.3 LTS

#### Guest operating system

Ubuntu 22.04.4 live server (amd64)

#### Steps to reproduce

packer init ./
packer build -var-file=variables.auto.pkvars.hcl ./

The variables file contains HCP client and secret. 

#### Packer template file

[Gist] (https://gist.github.com/agafonoo/6d18ef31f2eeb3923059a3d8acbeefa8)

### PR DESCRIPTION
**Pull request by chrisroberts**
If the registry box needs to be created, wait for the create operation to finish before allowing the post-processor to continue to the next step.

Closes #128

### CODE REVIEW

**Review by nywilken — COMMENTED** [review]
@chrisroberts thanks for putting together a fix so quickly. Overall the added wait looks good to me. I left a comment on the error message as I found them to be a bit cryptic. 

When it comes to working with HCP related things we found the errors that come back from the clients tend to lack a call to action. So we like to provide users with verbose error messages and provide them with a call to action if they can retry or do something to resolve the error elsewhere.

File: post-processor/hcp-vagrant-registry/step_create_box.go [hunk 1]
```diff
@@ -15,6 +17,8 @@ import (
 
 type stepCreateBox struct{}
 
+var BOX_CREATE_TIMEOUT = "60s"
```
[the remark this answers is not in the document]
Comment (chrisroberts) [reply post-processor/hcp-vagrant-registry/step_create_box.go:20]: This timeout is only for the creation of the box resource on HCP, and not the upload.

File: post-processor/hcp-vagrant-registry/step_create_box.go [hunk 2]
```diff
@@ -67,6 +71,48 @@ func (s *stepCreateBox) Run(ctx context.Context, state multistep.StateBag) multi
 	}
 
 	ui.Say(fmt.Sprintf("Created new box: %s", config.Tag))
+	if cresp.Payload == nil || cresp.Payload.Operation == nil {
+		state.Put("error", fmt.Errorf("Unable to wait for box to become available - invalid response received."))
```
[the remark this answers is not in the document]
Comment (chrisroberts) [reply post-processor/hcp-vagrant-registry/step_create_box.go:75]: A missing payload in the response, or the payload not including the operation _shouldn't happen_ if the response was successful, but we still want to test for the unexpected. In this case, there would be no operation to wait on but the response from the box creation would be successful. Would updating the error message to something like this work: "Unable to wait for box to become available - invalid response received. Please try again."

**Review by nywilken — APPROVED** [review]
Thanks for the interval discussion. I understand the approach and why you are choosing to halt and not continue with a warning to user. I left a suggestion to the error message but it may not help anymore than what you already have. 

I think in the end if the user knows what state the box is in and what they need to do next that should be enough to help them resolve any intermittent connectivity or timeout related blips.

**Pull request merged**